File: service/src/main/java/net/huray/backend/minuting/web/home/SpaceController.java
package net.huray.backend.minuting.web.home;

import java.util.List;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import net.huray.backend.minuting.common.dto.DataResponse;
import net.huray.backend.minuting.service.space.SpaceService;
import net.huray.backend.minuting.web.home.data.PublicSpaceRes;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequestMapping(value = "/space")
@RestController
@RequiredArgsConstructor
public class SpaceController {

    private final SpaceService spaceService;

    @GetMapping("/public")
    public DataResponse<List<PublicSpaceRes>> listPublicSpace() {
        UUID uid = UUID.fromString("55a03586-5f14-11ed-8e54-6cd5e80d8470");
        return DataResponse.from(spaceService.listPublicSpace(uid));
    }
}

File: service/src/main/kotlin/net/huray/backend/minuting/controller/SpaceController.kt
package net.huray.backend.minuting.controller

import net.huray.backend.http.res.DoneResult
import net.huray.backend.http.res.ListResult
import net.huray.backend.http.res.SingleResult
import net.huray.backend.minuting.contract.SpaceContract
import net.huray.backend.minuting.dto.SpaceDto
import net.huray.backend.minuting.service.SpaceService
import net.huray.backend.minuting.support.AuthenticationFacade
import org.springframework.web.bind.annotation.RestController

@RestController
class SpaceController(
    private val spaceService: SpaceService,
    private val authenticationFacade: AuthenticationFacade
) : SpaceContract {

    override fun create(req: SpaceDto.CreateReq) =
        SingleResult(spaceService.create(authenticationFacade.uid, req))

    override fun get(id: Long) =
        SingleResult(spaceService.get(authenticationFacade.uid, id))

    override fun listPublic() =
        ListResult(spaceService.listPublic(authenticationFacade.uid))

    override fun update(id: Long, req: SpaceDto.UpdateReq): DoneResult {
        spaceService.update(authenticationFacade.uid, id, req)

        return DoneResult()
    }

    override fun delete(id: Long) {
        spaceService.delete(authenticationFacade.uid, id)
    }

    override fun join(id: Long) {
        spaceService.join(authenticationFacade.uid, id)
    }

}